pdns_exporter/pdns_exporter.py

```diff
@@ -0,0 +1,119 @@
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in
+#    the documentation and/or other materials provided with the
+#    distribution.
+# 3. Neither the name of the Company nor the names of its contributors
+#    may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+#
+# ALTERNATIVELY, provided that this notice is retained in full, this
+# product may be distributed under the terms of the GNU General Public
+# License (GPL) version 2 or later, in which case the provisions
+# of the GPL apply INSTEAD OF those given above.
+#
+# This software is provided ``as is'', and any express or implied
+# warranties, including, but not limited to, the implied warranties of
+# merchantability and fitness for a particular purpose are disclaimed.
+# In no event shall the company or contributors be liable for any
+# direct, indirect, incidental, special, exemplary, or consequential
+# damages (including, but not limited to, procurement of substitute
+# goods or services; loss of use, data, or profits; or business
+# interruption) however caused and on any theory of liability, whether
+# in contract, strict liability, or tort (including negligence or
+# otherwise) arising in any way out of the use of this software, even
+# if advised of the possibility of such damage.
+
+import sys
+import pytrap
+import argparse
+import json
+import gzip
+import socket
+from datetime import datetime
+
+"""
+This module reads flow data from TRAP interface and export flow data to JSON and
+store JSON to files on disk. Data in JSON are in format to process in Passive DNS.
+"""
+
+parser = argparse.ArgumentParser(description='Module for exporting flow data to format for Passive DNS')
+parser.add_argument('-i',  "--ifcspec", help="select TRAP IFC specifier")
+args = parser.parse_args()
+
+context = pytrap.TrapCtx()
+context.init(['-i', args.ifcspec])
+
+context.setRequiredFmt(0, pytrap.FMT_UNIREC, "")
+unirec = pytrap.UnirecTemplate("ipaddr DST_IP,ipaddr SRC_IP,uint64 BYTES,uint64 LINK_BIT_FIELD,time TIME_FIRST, "\
+                               "time TIME_LAST,uint32 DNS_RR_TTL,uint32 PACKETS,uint16 DNS_ANSWERS,uint16 DNS_CLASS,"\
+                               "uint16 DNS_ID,uint16 DNS_PSIZE,uint16 DNS_QTYPE,uint16 DNS_RLENGTH,uint16 DST_PORT,"\
+                               "uint16 SRC_PORT,uint8 DIR_BIT_FIELD,uint8 DNS_DO,uint8 DNS_RCODE,"\
+                               "uint8 PROTOCOL,uint8 TCP_FLAGS,uint8 TOS,uint8 TTL,string DNS_NAME,bytes DNS_RDATA")
+
+records = list()
+CONTENT_THRESHOLD = 5000
+
+# ===== main lopp =====
+while True:
+    try:
+        data = context.recv()
+    except pytrap.FormatMismatch:
+        sys.stderr.write("Error: input data format mismatch.\n")
+        break
+    except pytrap.FormatChanged as e:
+        data = e.data
+        del(e)
+        pass
+    except pytrap.Terminated:
```

Comment:
This exception means all next recv() calls will fail. That's why `pass` seems to be bad idea.